### DISCUSSION TITLE
Multiple node computation

### DISCUSSION THREAD

**Discussion by jk262626** [nakib/elphbolt#143] [General]
Hi Nakib,
Would it be allowed to run the Elphbolt using multiple nodes?
Say for example, use 2 compute nodes with 50 cores per node.

**Comment by nakib** [comment]
Dear jk262626,

Of course. I think the largest computation I have ever done with this code was with 10 nodes x 36 cores/node. Note that since the code does a large amount of disk i/o, beyond a certain number of cores, whatever benefit you would get from parallelization might get diminished by the i/o latency. It is all system dependent and the best way to proceed is to first to do a benchmark on the cluster you are using. I'd appreciate it if you reported back your experience.

Best regards,
Nakib

**Reply by jk262626** [reply]
I'm a little bit confused about whether I should use 360 or 36 for the number of coarray images when I  use 10 nodes x 36 cores/node.
(in the input script, "cafrun -n 360 elphbolt" this line)

And another thing I wonder, what was the memory per node for your computation?
I am running (50, 150) mesh file with 1 node x 25 cores/node under the computation node with 256GB memory and fails to run.

**Reply by nakib** [reply]
The number of images should be 360. Every core is an image.

The amount of memory required depends on the sizes of the various wannierized quantities. I can't comment without knowing some of the details of the parameters of your calculation. For all of my calculations, the number of Wannier bands was quite small, so memory was never an issue.

Best,
Nakib

**Comment by jk262626** [comment]
I see, it is good to know. Can I use just a few Wannier band around
transport bands?


Best regards,

Jiyoung Kim


*----------------------------------------------------------------------------------------------*

*Jiyoung Kim*

Ph.D. Student

Sibley School of Mechanical and Aerospace Engineering| Cornell University
Ithaca, NY 14850 | ***@***.*** ***@***.***>


*Gender Pronouns: she, her, hers*

**Reply by nakib** [reply]
Yes. You already choose the number of transport active bands in elphbolt. But the number of Wannier bands included in all the wannierized files is what dictates how much space will be needed. You can check in this paper how the wannerized quantities scale in space: https://journals.aps.org/prb/pdf/10.1103/PhysRevB.76.165108

Best,
Nakib

**Comment by jk262626** [comment]
Awesome, thanks for the info!
